test/unit/waveform-builder-spec.js

```diff
@@ -193,6 +193,60 @@ describe('WaveformBuilder', function() {
       }).to.throw();
     });
 
+    ('ArrayBuffer' in window) && it('should throw if arraybuffer data is invalid', function(done) {
+      expect(function() {
```

Comment:
It seems strange for a function that takes a callback to throw errors. I realise you're following the existing convention in the codebase, which is good, but this is something that should be fixed, really. I'll take a look.

Reply:
You're right, I should have made some for `Peaks.init` as well. 
I see that you've changed the tests a bit. I will add those tests following your new format tomorrow.

Reply:
Thank you! I have made a change so that callbacks are used consistently for error handling, so please update your branch again.

Reply:
New tests created! Please let me know if anything else needs to be changed!